code/datums/mutations/touch.dm

```diff
@@ -51,7 +51,7 @@
 /datum/action/cooldown/spell/touch/shock/cast_on_hand_hit(obj/item/melee/touch_attack/hand, atom/victim, mob/living/carbon/caster)
 	if(iscarbon(victim))
 		var/mob/living/carbon/carbon_victim = victim
-		if(carbon_victim.electrocute_act(15, caster, 1, SHOCK_NOGLOVES | SHOCK_NOSTUN))//doesnt stun. never let this stun
+		if(carbon_victim.electrocute_act(15, caster, 1, SHOCK_NOGLOVES | SHOCK_NOSTUN) && !carbon_victim.has_status_effect(/datum/status_effect/his_grace))//doesnt stun. never let this stun MONKESTATION ADDITION: his grace check
```

Comment:
Maybe it should check for certain traits instead? i.e his grace adds all of the "pain immunity" traits (`TRAIT_ABATES_SHOCK`, `TRAIT_ANALGESIA`, `TRAIT_NO_PAIN_EFFECTS`, and `TRAIT_NO_SHOCK_BUILDUP`), altho I imagine there's more you could add, or maybe check for a more fitting trait, and add that to his_grace? or make a new trait.

Reply:
I was originally gonna do a trait check for one of the pain effect didn't really find any fitting since shocking isn't very pain related, it already respects anti drop and his grace was kinda a special case. Is checking for traits bad in some way?
I'll make its own trait if so, I mean its a small combo rare to come up in the first place

Reply:
It makes it more annoying to make other things immune to it in the future.

Reply:
decided on using TRAIT_NO_SHOCK_BUILDUP

Reply:
think I should add more than just one?

Reply:
perhaps also check for `TRAIT_SHOCKIMMUNE` and `TRAIT_STUNIMMUNE`?

Reply:
just added shockimmune, as far as I see nothing for checking for several traits in a list, gonna try to keep it short. It doesn't stun anyways